presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -346,7 +347,15 @@ public static ConnectorPageSource createParquetPageSource(
                         List<String> nestedColumnPath = nestedColumnPath(pushedDownSubfield);
                         Optional<ColumnIO> columnIO = findNestedColumnIO(lookupColumnByName(messageColumnIO, pushedDownSubfield.getRootName()), nestedColumnPath);
                         if (columnIO.isPresent()) {
-                            fieldsBuilder.add(constructField(type, columnIO.get()));
+                            if (isPushedDownSubfield(column)) {
+                                Type nestedType = getNestedStructType(pushedDownSubfield, type);
+                                GroupColumnIO newColumn = (GroupColumnIO) lookupColumnByName(messageColumnIO, pushedDownSubfield.getRootName());
+                                Optional<Field> newfield = constructField(nestedType, newColumn, true);
```

Comment:
Since you have written the line 357 as fieldsBuilder.add(constructField(...));, it would be consistent if you can do the same instead of splitting into separate lines.